src/specBuilder/bar/barUtils.ts

```diff
@@ -254,6 +256,44 @@ export const getStroke = ({ name, children, color, colorScheme }: BarSpecProps):
 	];
 };
 
+export const getDimensionSelectionRing = (props: BarSpecProps): RectMark => {
+	const { name, colorScheme, paddingRatio, orientation } = props;
+
+	const update =
+		orientation === 'vertical'
+			? {
+					y: { value: -5 },
+					y2: { signal: 'height + 5' },
+					xc: { signal: `scale('xBand', datum.${name}_selectedGroupId) + bandwidth('xBand')/2` },
+					width: { signal: `bandwidth('xBand')/(1 - ${paddingRatio} / 2)` },
+			  }
+			: {
+					x: { value: -5 },
+					x2: { signal: 'width + 5' },
```

Comment:
The 5 px added above/below the bar causes jitter when you open the popover. This is because you are adding marks that extend belong the chart area and that causes the whole chart to recalculate its size. I'm confirming with Alan but I think we want to drop this so that we don't get jitter.

https://github.com/user-attachments/assets/5ce86b48-aeed-4fb8-bd9f-0e95d24eac70